### PR TITLE
feat: update oz erc1967 + reward receiver + deployment script

### PR DESCRIPTION
**Pull request by kobe-eth**
# Features:

* Add ERC1967 Proxy from OZ to have verifiable contract on Etherscan
* Reward Receiver for Gauges that have "reward_receivers" logic.
* Migration script minus test addresses and empty strategies

### CODE REVIEW

File: src/strategy/only-boost/OnlyBoost.sol
Comment (PumpkingWok) [inline src/strategy/only-boost/OnlyBoost.sol:1]: In `acceptGovernance()` we may set the `futureGovernance` to address(0).

**Comment by Lao0ni** [comment]
**Typos:** 
- @ : https://github.com/stake-dao/only-boost/blob/1b039a94b3425677443aea7095c62d971873c6df/src/factory/curve/CRVPoolFactory.sol#L51

- Adding natspec comment on `SafeExecute.sol` , like saying why it is needed

- Addres => Address : https://github.com/stake-dao/only-boost/blob/1b039a94b3425677443aea7095c62d971873c6df/src/optimizer/Optimizer.sol#L113

- Strategie => Strategy : https://github.com/stake-dao/only-boost/blob/1b039a94b3425677443aea7095c62d971873c6df/src/optimizer/Optimizer.sol#L148

- For all transfer / accept governance mechanism, put same comments (different on Optimizer)

- Missing @param for distributeSDT => https://github.com/stake-dao/only-boost/blob/1b039a94b3425677443aea7095c62d971873c6df/src/strategy/only-boost/OnlyBoost.sol#L88

- Missing @params => https://github.com/stake-dao/only-boost/blob/1b039a94b3425677443aea7095c62d971873c6df/src/strategy/only-boost/OnlyBoost.sol#L190

**Comment by kobe-eth** [comment]
https://github.com/stake-dao/only-boost/blob/a5674966715197348386cd55d02557fa87eeec5d/src/optimizer/Optimizer.sol#L121

Remove comments // New

**Comment by kobe-eth** [comment]
https://github.com/stake-dao/only-boost/blob/a5674966715197348386cd55d02557fa87eeec5d/src/strategy/only-boost/OnlyBoost.sol#L177

Unchecked loop to remain consistent

**Comment by kobe-eth** [comment]
https://github.com/stake-dao/only-boost/blob/a5674966715197348386cd55d02557fa87eeec5d/src/factory/PoolFactory.sol#L34

Update with 0x08d36c723b8213122f678025C2D9eb1Ec7Ab8F9D

**Pull request merged**